### DISCUSSION TITLE
Accessing a component element inside onMounted hook when navigating on a page doesn't work

### DISCUSSION THREAD

**Discussion by kevhof** [nuxt/nuxt#17278] [Questions]
When you try to access an element from a component like this :
``` javascript
<script setup>
  onMounted(()=>{
    console.log(document.querySelector(".element"))
  })
</script>
```

It works only if you load the website on the page where there is the component. If you load the website on another page and then move to the page with the component, you get `null`.

This doesn't work too :
``` javascript
<script setup>
  onMounted(()=>{
    nextTick(()=>{
      console.log(document.querySelector(".element"))
    })
  })
</script>
```

This is the only solution at the moment : 
``` javascript
<script setup>
  onMounted(()=>{
    setTimeout(()=>{
      console.log(document.querySelector(".element"))
    }, 50) // 0 doesn't work either...
  })
</script>
```

**Comment by Zibbp** [comment]
Ran into the same issue. Using a ref to access the element does not work either.
```
<div ref="testDiv">test</div>
```
```
const testDiv = ref(null)
```

Using `setTimeout` allows me to access the element using any method.

**Reply by olivervorasai** [reply]
This bug is being tracked here https://github.com/nuxt/framework/issues/3587
